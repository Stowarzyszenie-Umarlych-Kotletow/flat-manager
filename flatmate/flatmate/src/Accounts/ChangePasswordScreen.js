import {Text, TextInput, View} from "react-native";
import styles from "../native_elements_styles";
import {Button} from "react-native-elements";
import * as React from "react";
import '../bootstrap_style.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.js';

export function ChangePasswordScreen({navigation}) {
    function onChangePasswordPress() {
        console.log('saving data')
        navigation.pop()
    }

    return (
        <View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Change password</Text>

                <TextInput placeholder="Old password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>

                <TextInput placeholder="New password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>

                <TextInput placeholder="Retype password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>

                <Button
                    buttonStyle={styles.bluButton}
                    onPress={() => onChangePasswordPress()}
                    title="Save"
                />
            </View>
        </View>
    );
}