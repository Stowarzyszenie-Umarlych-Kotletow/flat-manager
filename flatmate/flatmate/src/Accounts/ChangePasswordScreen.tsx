import {Text, TextInput, View} from "react-native";
import styles from "../static/styles";
import {Button} from "react-native-elements";
import {Controller, useForm} from "react-hook-form";
import * as React from "react";
import {useState} from "react";
import {useAppDispatch} from "../store";
import {ChangePasswordRequest} from "../models/api/account";


export function parseData(data) {
    const parsedData: ChangePasswordRequest = {
        "password": data.newPassword,
    }
    return parsedData;
}

export function ChangePasswordScreen() {
    const {control, handleSubmit, formState: {errors}} = useForm({
        defaultValues: {
            currentPassword: '',
            newPassword: '',
            newPasswordConfirm: '',
        },
    });
    const [showIncorrectPasswordWarning, setShowIncorrectPasswordWarning] = useState(false);
    const [showDifferentPasswordsWarning, setShowDifferentPasswordsWarning] = useState(false);
    
    const dispatch = useAppDispatch();

    async function onChangePasswordPress(data) {

        if (data.newPassword != data.newPasswordConfirm || !data.newPassword) {
            setShowDifferentPasswordsWarning(true);
        } else {
            setShowDifferentPasswordsWarning(false);
        }
        let parsedData = parseData(data);
        // backend connection
        // change password
        setShowIncorrectPasswordWarning(false);
        // todo
    }

    return (
        <View style={styles.accScreenContainer}>
            <Text style={styles.logoText}>Change password</Text>

            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Current password"
                        secureTextEntry={true}
                    />
                )}
                name="currentPassword"
            />

            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="New password"
                        secureTextEntry={true}
                    />
                )}
                name="newPassword"
            />

            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Confirm new password"
                        secureTextEntry={true}
                    />
                )}
                name="newPasswordConfirm"
            />

            {!showIncorrectPasswordWarning ? null :
                <Text style={styles.warningText}> Incorrect Current Password</Text>}
            {!showDifferentPasswordsWarning ? null :
                <Text style={styles.warningText}> Passwords do not match <br/> or <br/>You are trying to push empty
                    password </Text>}
            <Button
                buttonStyle={styles.greenButton}
                title="Change Password"
                onPress={handleSubmit(onChangePasswordPress)}
            />
        </View>
    );
}