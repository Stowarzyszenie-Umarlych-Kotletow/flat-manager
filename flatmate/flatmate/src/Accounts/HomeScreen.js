import {Text, View} from 'react-native';
import {Button} from 'react-native-elements';
import * as React from "react";
import styles from "../static/native_elements_styles";

export function HomeScreen({
                               navigation
                           }) {
    return (<View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Flatmate</Text>

                <Button
                    buttonStyle={styles.bluButton}
                    title="Login"
                    onPress={() => navigation.navigate('LoginScreen')}
                />

                <Button
                    buttonStyle={styles.bluButton}
                    title="Register"
                    onPress={() => navigation.navigate('RegisterScreen')}
                />
            </View>
        </View>);
}