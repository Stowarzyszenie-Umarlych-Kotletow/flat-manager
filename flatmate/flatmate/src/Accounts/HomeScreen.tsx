import {Text, View} from 'react-native';
import {Button} from 'react-native-elements';
import * as React from "react";
import styles from "../static/styles";

export function HomeScreen({navigation}) {
    return (<View style={styles.accScreenContainer}>
        <Text style={styles.loginLogoText}>Flatmate</Text>

        <Button
            buttonStyle={styles.blueButton}
            title="Login"
            onPress={() => navigation.navigate('LoginScreen')}
        />

        <Button
            buttonStyle={styles.blueButton}
            title="Register"
            onPress={() => navigation.navigate('RegisterScreen')}
        />
    </View>);
}