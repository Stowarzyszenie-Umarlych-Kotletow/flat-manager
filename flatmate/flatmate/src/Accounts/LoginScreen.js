import {Text, TextInput, View} from "react-native";
import styles from "../static/native_elements_styles";
import {Button} from "react-native-elements";
import * as React from "react";

export function LoginScreen({navigation, setUser}) {
    function onLoginPress() {
        console.log('test')
        setUser(123)
    }

    return (
        <View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Flatmate</Text>

                <TextInput placeholder="Username or e-mail" placeholderColor="#c4c3cb" style={styles.accFormTextInput}/>

                <TextInput placeholder="Password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>

                <Button
                    buttonStyle={styles.bluButton}
                    onPress={() => onLoginPress()}
                    title="Login"
                />
            </View>
        </View>
    );
}