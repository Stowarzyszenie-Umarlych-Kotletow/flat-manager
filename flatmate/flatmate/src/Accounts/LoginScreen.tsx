import {Text, TextInput, View} from "react-native";
import styles from "../static/styles";
import {Button} from "react-native-elements";
import {Controller, useForm} from "react-hook-form";
import * as React from "react";
import {useAppDispatch} from "../store";
import {LoginRequest} from "../models/api/auth";
import {login} from "../features/auth";

export function parseData(data) {
    const parsedData: LoginRequest = {
        "username": data.username,
        "password": data.password,
    }
    return parsedData;
}

export function LoginScreen() {
    const {control, handleSubmit, formState: {errors}} = useForm({
        defaultValues: {
            username: '',
            password: '',
        },
    });
    
    const dispatch = useAppDispatch();

    async function onLoginPress(data) {
        let parsedData = parseData(data);
        dispatch(login(parsedData))
            .unwrap()
            .then(() => {
                console.log("Logged in!");
            }, (err) => {
                console.log("Error!");
                console.log(err);
            });
    }

    return (
        <View style={styles.accScreenContainer}>
            <Text style={styles.loginLogoText}>Flatmate</Text>
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Username"
                    />
                )}
                name="username"
            />
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({field: {onChange, onBlur, value}}) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Password"
                        secureTextEntry={true}
                    />
                )}
                name="password"
            />
            <Button
                buttonStyle={styles.blueButton}
                title="Submit"
                onPress={handleSubmit(onLoginPress)}
            />
        </View>
    );
}