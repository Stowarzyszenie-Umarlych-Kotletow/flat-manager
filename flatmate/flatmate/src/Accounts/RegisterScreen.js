import {Text, TextInput, View} from "react-native";
import styles from "../native_elements_styles";
import {Button} from "react-native-elements";
import * as React from "react";

export function RegisterScreen() {
    function onRegisterPress() {
        console.log('test')
    }

    return (
        <View style={styles.accScreenContainer}>
            <View style={styles.accFormView}>
                <Text style={styles.logoText}>Flatmate</Text>
                <TextInput placeholder="Username" placeholderColor="#c4c3cb" style={styles.accFormTextInput}/>
                <TextInput placeholder="E-mail" placeholderColor="#c4c3cb" style={styles.accFormTextInput}/>
                <TextInput placeholder="Password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>
                <TextInput placeholder="Retype password" placeholderColor="#c4c3cb" style={styles.accFormTextInput}
                           secureTextEntry={true}/>
                <Button
                    buttonStyle={styles.bluButton}
                    onPress={() => onRegisterPress()}
                    title="Register"
                />
            </View>
        </View>
    );
}