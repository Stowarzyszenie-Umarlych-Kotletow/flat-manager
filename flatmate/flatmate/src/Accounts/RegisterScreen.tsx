import { Text, TextInput, View } from "react-native";
import styles from "../static/styles";
import { Button } from "react-native-elements";
import { Controller, useForm } from "react-hook-form";
import * as React from "react";
import { yupResolver } from '@hookform/resolvers/yup';
import * as Yup from "yup";
import {useAppDispatch, useAppSelector} from "../store";
import { useRegisterMutation } from "../features/api/user-api";
import { RegisterRequest } from "../models/api/auth";

export function parseRegisterData(data): RegisterRequest {
    let parsedData = {
        "firstName": data.firstName,
        "lastName": data.lastName,
        "username": data.username,
        "email": data.email,
        "password": data.password,
    }
    return parsedData
}

export function RegisterScreen({ navigation }) {
    const schema = Yup.object({
        password: Yup.string().required("Please enter your password")
            .matches(
                /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/,
                "Must Contain 8 Characters, One Uppercase, One Lowercase, One Number and one special case Character"
            ),
        confirmPassword: Yup.string().test("passwords-match", "Passwords must match", function (value) {
            return this.parent.password === value;
        }),
        email: Yup.string().required("Please enter your email").matches(/@/, "Incorrect email")
    }).required();

    const toastMessage = useAppSelector((state) => state.toast.message?.toString());

    const { control, handleSubmit, formState: { errors } } = useForm({
        defaultValues: {
            firstName: '',
            lastName: '',
            email: '',
            username: '',
            password: '',
            confirmPassword: '',
        },
        resolver: yupResolver(schema)
    });
    const [register, { isSuccess }] = useRegisterMutation();
    const onSubmit = data => onRegisterPress(data)

    function onRegisterPress(data) {
        let parsedData = parseRegisterData(data);
        register(parsedData).unwrap().then(
            () => {
                // TODO: add success toast
                navigation.goBack();
            }
        );
    }


    return (
        <View style={styles.accScreenContainer}>
            <Text style={styles.logoText}>Flatmate</Text>
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({ field: { onChange, onBlur, value } }) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="First Name"
                    />
                )}
                name="firstName"
            />
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({ field: { onChange, onBlur, value } }) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Last Name"
                    />
                )}
                name="lastName"
            />
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({ field: { onChange, onBlur, value } }) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Email"
                    />
                )}
                name="email"
            />
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({ field: { onChange, onBlur, value } }) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Username"
                    />
                )}
                name="username"
            />
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({ field: { onChange, onBlur, value } }) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Password"
                        secureTextEntry={true}
                    />
                )}
                name="password"
            />
            <Controller
                control={control}
                rules={{
                    maxLength: 100,
                }}
                render={({ field: { onChange, onBlur, value } }) => (
                    <TextInput
                        style={styles.accFormTextInput}
                        onBlur={onBlur}
                        onChangeText={onChange}
                        value={value}
                        placeholder="Confirm Password"
                        secureTextEntry={true}

                    />
                )}
                name="confirmPassword"
            />
            <Button
                buttonStyle={styles.blueButton}
                title="Submit"
                onPress={handleSubmit(onSubmit)}
            />
            <Text> {errors.email?.message} </Text>
            <Text> {errors.password?.message} </Text>
            <Text> {errors.confirmPassword?.message} </Text>
            <Text> {toastMessage} </Text>
        </View>
    );
}