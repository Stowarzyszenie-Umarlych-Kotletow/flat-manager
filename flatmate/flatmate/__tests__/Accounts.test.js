import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { mockStore } from "./utils/test-utils";
import store from "../src/store";
import { ModalPortal } from 'react-native-modals';
import { parseRegisterData, RegisterScreen } from '../src/screens/accounts/RegisterScreen'
import { LoginScreen, parseLoginData } from '../src/screens/accounts/LoginScreen';
import { HomeScreen } from "../src/screens/accounts/HomeScreen";
import { ChangePasswordScreen } from "../src/screens/accounts/ChangePasswordScreen";
import { ManageAccountScreen } from "../src/screens/accounts/ManageAccountScreen"

test('Parse register data', () => {
    let data = parseRegisterData({
        firstName: "test",
        lastName: "test",
        username: "test",
        email: "test",
        password: "test"
    });
    expect(data).toStrictEqual({
        "firstName": "test",
        "lastName": "test",
        "username": "test",
        "email": "test",
        "password": "test"
    });
});

test('Parse login data', () => {
    let data = parseLoginData({username: "test", password: "test"});
    expect(data).toStrictEqual({"username": "test", "password": "test"});
});

test('RegisterScreen renders correctly', () => {
    const tree = renderer.create(<Provider store={store}><RegisterScreen/> </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('HomeScreen renders correctly', () => {
    const tree = renderer.create(<Provider store={mockStore()}><HomeScreen /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('LoginScreen renders correctly', () => {
    const tree = renderer.create(<Provider store={store} ><LoginScreen /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('ChangePassword Screen renders correctly', () => {
    const tree = renderer.create(<Provider store={mockStore()}><ChangePasswordScreen /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('ManageAccountScreen Screen renders correctly', () => {
    const tree = renderer.create(<Provider store={mockStore()}><ManageAccountScreen /> <ModalPortal/> </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});

