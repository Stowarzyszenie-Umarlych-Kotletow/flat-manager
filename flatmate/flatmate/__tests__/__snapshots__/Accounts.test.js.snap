// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangePassword Screen renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "800",
        "marginBottom": 1,
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    Change password
  </Text>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Current password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#fafafa",
        "borderColor": "#eaeaea",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 14,
        "height": 43,
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingLeft": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="New password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#fafafa",
        "borderColor": "#eaeaea",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 14,
        "height": 43,
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingLeft": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Confirm new password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#fafafa",
        "borderColor": "#eaeaea",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 14,
        "height": 43,
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingLeft": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5cd07f",
            "borderColor": "#2089dc",
            "borderRadius": 5,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Change Password
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`HomeScreen renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "800",
        "marginBottom": 30,
        "marginTop": 150,
        "textAlign": "center",
      }
    }
  >
    Flatmate
  </Text>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3897f1",
            "borderColor": "#2089dc",
            "borderRadius": 5,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3897f1",
            "borderColor": "#2089dc",
            "borderRadius": 5,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Register
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`LoginScreen renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "fontSize": 40,
        "fontWeight": "800",
        "marginBottom": 30,
        "marginTop": 150,
        "textAlign": "center",
      }
    }
  >
    Flatmate
  </Text>
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Username"
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "#fafafa",
        "borderColor": "#eaeaea",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 14,
        "height": 43,
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingLeft": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onBlur={[Function]}
    onChangeText={[Function]}
    placeholder="Password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "backgroundColor": "#fafafa",
        "borderColor": "#eaeaea",
        "borderRadius": 5,
        "borderWidth": 1,
        "fontSize": 14,
        "height": 43,
        "marginBottom": 5,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 5,
        "paddingLeft": 10,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3897f1",
            "borderColor": "#2089dc",
            "borderRadius": 5,
            "borderWidth": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "center",
            "marginLeft": 25,
            "marginRight": 25,
            "marginTop": 10,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ManageAccountScreen Screen renders correctly 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "800",
          "marginBottom": 1,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Manage account
    </Text>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3897f1",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 10,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Log out
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#e74040",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 10,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Delete account
          </Text>
        </View>
      </View>
    </View>
  </View>,
  " ",
  " ",
]
`;

exports[`RegisterScreen renders correctly 1`] = `
Array [
  <View>
    <Text
      style={
        Object {
          "fontSize": 40,
          "fontWeight": "800",
          "marginBottom": 1,
          "marginTop": 20,
          "textAlign": "center",
        }
      }
    >
      Flatmate
    </Text>
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="First Name"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#eaeaea",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Last Name"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#eaeaea",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Email"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#eaeaea",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Username"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#eaeaea",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#eaeaea",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Confirm Password"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#eaeaea",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 14,
          "height": 43,
          "marginBottom": 5,
          "marginLeft": 5,
          "marginRight": 5,
          "marginTop": 5,
          "paddingLeft": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3897f1",
              "borderColor": "#2089dc",
              "borderRadius": 5,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "marginLeft": 25,
              "marginRight": 25,
              "marginTop": 10,
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>,
  " ",
]
`;
